### PR TITLE
feature / Refactor Interactive Charts to React

### PR DESCRIPTION
**Pull request by tehcoderer**
# Description

- [ ] Summary of the change / bug fix.
- [ ] Link # issue, if applicable.
- [ ] Screenshot of the feature or the bug before/after fix, if applicable.
- [ ] Relevant motivation and context.
- [ ] List any dependencies that are required for this change.

# How has this been tested?

* Please describe the tests that you ran to verify your changes.
* Provide instructions so we can reproduce.
* Please also list any relevant details for your test configuration.
- [ ] Make sure affected commands still run in terminal
- [ ] Ensure the SDK still works
- [ ] Check any related reports

# Checklist:

- [ ] I have adhered to the GitFlow naming convention and my branch name is in the format of `feature/feature-name` or `hotfix/hotfix-name`.
- [ ] Update [our documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).  Update any user guides that are affected by the changes.
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [integration test script](https://github.com/OpenBB-finance/OpenBBTerminal/tree/develop/openbb_terminal/miscellaneous/integration_tests_scripts).

# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by jose-donato** [comment]
🍝

**Comment by jose-donato** [comment]
> Are we moving to React from Plotly?

still using plotly

**Comment by tehcoderer** [comment]
> Are we moving to React from Plotly?

@the-praxs still using Plotly just refactored chart frontend from [JS ](https://github.com/OpenBB-finance/OpenBBTerminal/tree/develop/openbb_terminal/core/plots/web) to react

**Comment by the-praxs** [comment]
@tehcoderer @jose-donato gotcha!

**Comment by jmaslek** [comment]
Still no file popup when export on mac :(

File: frontend-components/plotly/src/components/Chart.tsx [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+      setModal,
+      setOnAnnotationClick,
+      setAnnotations,
+      onAnnotationClick,
+      annotations,
+      plotDiv,
+    });
+  }
+
+  if (axesTitles && Object.keys(axesTitles).length > 0) {
```
Comment (jose-donato) [inline frontend-components/plotly/src/components/Chart.tsx:99]: when should this run?

if it's when the component mounts it should run inside a useEffect(() => {
//code....    
},[])

File: frontend-components/plotly/src/components/Chart.tsx [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+        setLoading={setLoading}
+        open={modal.modal === "overlayChart"}
+        close={onClose}
+      />
+      <TitleChartDialog
+        updateTitle={(title) => setChartTitle(title)}
+        updateAxesTitles={(axesTitles) => setAxesTitles(axesTitles)}
+        defaultTitle={chartTitle}
+        plotlyData={plotData}
+        open={modal.modal === "titleDialog"}
```
[the remark this answers is not in the document]
Comment (tehcoderer) [reply frontend-components/plotly/src/components/Chart.tsx:397]: lmao forgot to rename it to modal.name

File: frontend-components/plotly/src/components/Chart.tsx (hunk 1 above)
Comment (tehcoderer) [reply frontend-components/plotly/src/components/Chart.tsx:99]: EZPZ

File: frontend-components/plotly/src/components/Chart.tsx [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+
+  const [barButtons, setModeBarButtons] = useState({});
+  const [LogYaxis, setLogYaxis] = useState(false);
+  const [chartTitle, setChartTitle] = useState(title);
+  const [axesTitles, setAxesTitles] = useState({});
+  const [plotLoaded, setPlotLoaded] = useState(false);
+  const [modal, setModal] = useState({ modal: "" });
+  const [loading, setLoading] = useState(false);
+  const [plotDiv, setPlotDiv] = useState(null);
+  const [Volume, setVolume] = useState({ old_nticks: {} });
```
[the remark this answers is not in the document]
Comment (tehcoderer) [reply frontend-components/plotly/src/components/Chart.tsx:57]: EZPZ

File: frontend-components/plotly/src/components/Chart.tsx [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+  json: Figure;
+  date: Date;
+  cmd: string;
+  title: string;
+  global: any;
+  info?: any;
+}) {
+  const posthog = usePostHog();
+
+  if (info)
```
[the remark this answers is not in the document]
Comment (tehcoderer) [reply frontend-components/plotly/src/components/Chart.tsx:38]: EZPZ

**Comment by andrewkenreich** [comment]
@tehcoderer We need to clear the box if the user hits cancel on Windows

File: openbb_terminal/core/log/generation/settings_logger.py [hunk 5]
```diff
@@ -57,11 +57,26 @@ def log_credentials() -> None:
         else:
             current_keys[cfg_var_name] = "not_defined"
 
-    logger.info("KEYS: %s ", json.dumps(current_keys))
+    # logger.info("KEYS: %s ", json.dumps(current_keys))
+    return current_keys
 
 
 def log_preferences() -> None:
```
Comment (hjoaquim) [inline openbb_terminal/core/log/generation/settings_logger.py:64]: This function should be removed

**Review by jmaslek — APPROVED** [review]

**Pull request merged**